server/events/plan_command_runner_test.go

```diff
@@ -508,3 +508,99 @@ func TestPlanCommandRunner_ExecutionOrder(t *testing.T) {
 		})
 	}
 }
+
+func TestPlanCommandRunner_AtlantisApplyStatus(t *testing.T) {
+	logger := logging.NewNoopLogger(t)
+	RegisterMockTestingT(t)
+
+	cases := []struct {
+		Description            string
+		PrevPlanStored         bool // stores a "No Changes" plan in the backend
+		Targeted               bool
+		ExpVCSApplyStatusTotal int
+		ExpVCSApplyStatusSucc  int
+	}{
+		{
+			Description:            "When planning with no previous plan, set the atlantis/apply VCS status to 0/0",
+			ExpVCSApplyStatusTotal: 0,
+			ExpVCSApplyStatusSucc:  0,
+		},
+		{
+			Description:            "When planning with the previous plan that results in No Changes, set the atlantis/apply VCS status to 1/1",
+			PrevPlanStored:         true,
+			ExpVCSApplyStatusTotal: 1,
+			ExpVCSApplyStatusSucc:  1,
+		},
+		{
+			Description:            "When planning with unmatched target and any previous apply's, maintain VCS status",
+			PrevPlanStored:         true,
+			Targeted:               true,
+			ExpVCSApplyStatusTotal: 1,
+			ExpVCSApplyStatusSucc:  1,
+		},
```

Comment:
I feel like we are missing some conditions here that we could be testing for. What happens when plans override each other with new "No changes" or new "Pending" based on drift?

Reply:
Thanks @GenPage, I've added various test cases to cover different scenarios based on the review feedback. Please check!